website_event_sale_check_order_before_payment/controllers/main.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+from odoo import _, http
+from odoo.http import request
+
+import odoo.addons.website_sale.controllers.main as website_sale
+
+_logger = logging.getLogger(__name__)
+
+
+class WebsiteSale(website_sale.WebsiteSale):
+    @http.route(
+        ["/shop/payment/order_validity"], type="json", auth="public", website=True
+    )
+    def check_order_validity(self):
+        order = request.website.sale_get_order()
+        if order:
+            if order.state != "draft":
```

Comment:
Is it really cancelled if the `order.state != "draft"`?
Couldn't the order be `sent`, for example?

What's wrong with actually checking if it's cancelled? Are you worried about the other states?
```suggestion
            if order.state == "cancel":
```

Reply:
You're right, checking cancel state should be enough